### DISCUSSION TITLE
Parte 1: Diff minima en output de test hard y extreme

### DISCUSSION THREAD

**Discussion by GerardoAlcalde** [IIC2133-PUC/2024-2#94] [Tarea 1] [answered]
Problema con la Identificación del Pingüino más Feliz en Eventos Festival
He encontrado un problema al ejecutar casos de prueba grandes (20,000 líneas) en los eventos de tipo Festival. Específicamente, el ID del pingüino más feliz a veces es incorrecto en el resultado, aunque debería coincidir con el último pingüino en el arreglo ordenado por felicidad.

A continuación se muestran los resultados de diff entre el output_correcto.txt y el output.txt para dos casos: Hard y Extreme.

Caso Hard:
```vscode ➜ /workspaces/T1-2024-2-GerardoAlcalde (master) $ diff -b output_correcto.txt output.txt
3042c3042
<               81 14 164
---
>         81 142 164
3074c3074
<               81 14 164
---
>         81 142 164
3105c3105
<               81 14 164
---
>         81 142 164
3139c3139
<               81 14 164
---
>         81 142 164
...
6694c6694
<               130 5 390
---
>         130 130 390
```

Caso Extreme:
```vscode ➜ /workspaces/T1-2024-2-GerardoAlcalde (master) $ diff -b output_correcto.txt output.txt
6274c6274
<               84 18 20
---
>         84 150 20
6356c6356
<               84 18 20
---
>         84 150 20
6438c6438
<               84 18 20
---
>         84 150 20
...
85330c85330
<               133 383 705
---
>         133 659 705
```
Posibles Causas:
Error en la ordenación: Aunque uso qsort para ordenar los pingüinos por felicidad después de cada inserción y actualización, parece que en algunos casos el orden del arreglo puede ser incorrecto.
Condiciones de carrera (race conditions): En volúmenes muy altos de datos, como con 20,000 líneas de input, podría estar ocurriendo algún problema de sobrescritura o acceso indebido a memoria.

Cualquier ayuda o sugerencia sería muy apreciada.

**Comment by Segismundios** [comment, marked as answer]
Hola @GerardoAlcalde. Te recomendaría revisar tu planteamiento del problema. Es probable que qsort te esté dando problemas y, más allá de eso, revisa bien la complejidad de la operación qsort y si ésta cumple con lo pedido en el enunciado. Recuerda que para que tu solución sea considerada correcta, debes cumplir con la complejidad solicitada.
Saludos!
